### PR TITLE
[SPARK-40264][ML][DOCS] Supplement docstring in pyspark.ml.functions.predict_batch_udf

### PR DESCRIPTION
**Pull request by leewyang**
### What changes were proposed in this pull request?

Followup edits to pydoc from #37734 per [request](https://github.com/apache/spark/pull/37734#issuecomment-1384092385)

### Why are the changes needed?

Adds examples of multiple output models (using StructType return_type) for:
- returning multiple fields of array type
- returning a list of dict for row-oriented 

### Does this PR introduce _any_ user-facing change?

Yes, updated pydocs for `predict_batch_udf` function.

### How was this patch tested?

Documentation only.

### CODE REVIEW

**Comment by leewyang** [comment]
@WeichenXu123 Here's the followup PR w/ additional examples in the pydoc.

File: python/pyspark/ml/functions.py [hunk 1]
```diff
@@ -486,6 +486,7 @@ def predict(inputs: np.ndarray) -> np.ndarray:
     data = np.arange(0, 1000, dtype=np.float64).reshape(-1, 4)
```
Comment (HyukjinKwon) [inline python/pyspark/ml/functions.py:486]: Can we follow https://numpydoc.readthedocs.io/en/latest/format.html and add the examples as a proper doctest?

Comment (HyukjinKwon) [inline python/pyspark/ml/functions.py:486]: `predict_batch_udf` should also be added to https://github.com/apache/spark/blob/master/python/docs/source/reference/pyspark.ml.rst#functions

File: python/pyspark/ml/functions.py [hunk 2]
```diff
@@ -637,6 +641,62 @@ def predict_columnar(x1: np.ndarray, x2: np.ndarray) -> Mapping[str, np.ndarray]
...
+    df.withColumn("x2", multi_times_two_udf("t1", "t2")).select("t1", "t2", "x2.*").show(5)
+    # +--------------------+------------------+--------------------+------------------+
+    # |                  t1|                t2|                t1x2|              t2x2|
+    # +--------------------+------------------+--------------------+------------------+
+    # |[0.0, 1.0, 2.0, 3.0]|   [0.0, 1.0, 2.0]|[0.0, 2.0, 4.0, 6.0]|   [0.0, 2.0, 4.0]|
+    # |[4.0, 5.0, 6.0, 7.0]|   [4.0, 5.0, 6.0]|[8.0, 10.0, 12.0,...| [8.0, 10.0, 12.0]|
+    # |[8.0, 9.0, 10.0, ...|  [8.0, 9.0, 10.0]|[16.0, 18.0, 20.0...|[16.0, 18.0, 20.0]|
+    # |[12.0, 13.0, 14.0...|[12.0, 13.0, 14.0]|[24.0, 26.0, 28.0...|[24.0, 26.0, 28.0]|
+    # |[16.0, 17.0, 18.0...|[16.0, 17.0, 18.0]|[32.0, 34.0, 36.0...|[32.0, 34.0, 36.0]|
+    # +--------------------+------------------+--------------------+------------------+
```
Comment (HyukjinKwon) [inline python/pyspark/ml/functions.py:699]: backtick doesn't work in sphinx IIRC. I meant ```` ``` ````

File: python/pyspark/ml/functions.py [hunk 3]
```diff
@@ -647,37 +386,369 @@ def predict_columnar(x1: np.ndarray, x2: np.ndarray) -> Mapping[str, np.ndarray]
...
-    A pandas_udf for predicting a batch.
+    :py:class:`UserDefinedFunctionLike`
+        A Pandas UDF for model inference on a Spark DataFrame.
+
+    Examples
+    --------
+    For a pre-trained TensorFlow MNIST model with two-dimensional input images represented as a
+    flattened tensor value stored in a single Spark DataFrame column of type `array<float>`.::
+
+        from pyspark.ml.functions import predict_batch_udf
```
Comment (HyukjinKwon) [inline python/pyspark/ml/functions.py:432]: If this example cannot be ran, should better do it with `.. code-block:: python`

**Review by HyukjinKwon — APPROVED** [review]
Looks fine from a cursory look

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**